### ISSUE TITLE
Unable to run basic django app on wsl2 vscode cloud run emulator

### ISSUE THREAD

**Issue by kunlegiwa** [GoogleCloudPlatform/cloud-code-vscode#338]
**Environment**: 
*(Hint: "Report Extension Issue on Github" command will fill these out for you.)*
Extension version:
VSCode version:
OS: Windows wsl2

**Description**: Unable to run the basic django app that comes with the cloud code extension on vscode

**Repro step**: 
Set up wsl2 on windows 10
Choose Ubuntu 20.04 as distro
Launch VSCode on Ubuntu and add cloud code extension
Create a new cloud run application from vscode
Run on cloud run emulator and the error below is thrown

Dependency check started
Dependency check succeeded
Starting minikube, this may take a while....................................................
minikube successfully started

creating runner: creating builder: getting docker client: getting minikube env: running [/mnt/c/ProgramData/chocolatey/bin/minikube.exe docker-env --shell none -p cloud-run-dev-internal]
 - stdout: "\n\n"
 - stderr: "❌  Exiting due to RUNTIME_RESTART: The Docker service within 'cloud-run-dev-internal' is not active\n"
 - cause: exit status 90
. If above error is unexpected, please open an issue https://github.com/GoogleContainerTools/skaffold/issues/new to report this error.
Exited with code 1.
Cleaning up...
Finished clean up.

**Comment by vincentjocodes** [comment]
@kunlegiwa Thanks for the report, we are going to try to repro this bug.

**Comment by sivakku** [comment]
We are working on investigating this further on wsl2.

**Comment by quoctruong** [comment]
I was able to repro this issue:

```
quoct@quoct0-w:~$ minikube docker-env --shell none -p cloud-run-dev-internal
DOCKER_TLS_VERIFY=1
DOCKER_HOST=tcp://127.0.0.1:32794
DOCKER_CERT_PATH=/home/quoct/.minikube/certs
MINIKUBE_ACTIVE_DOCKERD=cloud-run-dev-internal
quoct@quoct0-w:~$ /mnt/c/Users/quoct/AppData/Local/Google/Cloud\ SDK/google-cloud-sdk/bin/minikube.exe profile lis^C
quoct@quoct0-w:~$ /mnt/c/Users/quoct/AppData/Local/Google/Cloud\ SDK/google-cloud-sdk/bin/minikube.exe docker-env --shell none -p cloud-run-dev-internal
* There is no local cluster named "cloud-run-dev-internal"
  - To fix this, run: "minikube start -p cloud-run-dev-internal"
 ```

@medyagh
It seems like Cloud Code was using the `minikube.exe` instead of `minikube` on WSL2 and this is what causes the issue. Windows seems to allow users to run `.exe` file even in WSL2.

**Comment by medyagh** [comment]
thanks to the live debug meeting we had with  @quoctruong the root cause is, on WSL. (ubuntu) vs code still runs the Windows binary of minikube which looks for the minikube home in a different folder.

on minikube side should at least detect that the Windows binary is being executed in Linux env and warn the user or do the right thing. I will create an issue on minikube to handle WSL2 using the .exe binary better. 

meanwhile I suggest that cloud code on WSL2 uses the linux binary. this does add a bit of complexity which means if user switch between WSL2 and Powershell there would be two minikube homes.

**Comment by go-nelango** [comment]
@medyagh could you please link the minikube issue as well?

**Comment by medyagh** [comment]
I believe this issue was fixed in Skaffold 1.18.0 as I see in the release notes
https://github.com/GoogleContainerTools/skaffold/releases/tag/v1.18.0

on minikube side we could provide a warnning to the user that they might be running the wrong binary in WSL

**Comment by sivakku** [comment]
This issue is addressed in skaffold v1.18.0 release([#5085](https://github.com/GoogleContainerTools/skaffold/issues/5085)), if the VSCode has auto dependencies setting turned on then it will upgrade to the latest version of skaffold. Closing this issue as fixed.
